### ISSUE TITLE
Erorr012 after folder moving

### ISSUE THREAD

**Issue by piotr5151** [marcelklehr/floccus#423]
After cleaning and moving some folders on my laptop I switched to my desktop PC and I've received Error012.
My laptop still syncs perfectly, but desktop doesn't.

I tried to investigate the conditions of the Error012 and I've found that floccus is comparing number of folder's children locally with server.  

```
if (order.length !== folder.children.length) {
      throw new Error(browser.i18n.getMessage('Error012'))
 }
```

This is something I don't understand. After cleaning and moving folders of course I've change the numer and order of children. Does it means that removing folders aren't allowed ?

How can I fix it ? And how to avoid such problem in the future ?

**Comment by marcelklehr** [comment]
Hello @piotr5151 
Thanks for your feedback! I take it you are using the WebDAV backend?
The error you are seeing is more intricate than a mere mismatch of local vs. server folder length, it's a mismatch between the ordering matrix and the current state of the folder in the cache, which is actually a bug in the code. Would you be amendable to providing debug logs at the following share: https://klehranlage.net/index.php/s/7nn82NSBsfQKdsJ

**Comment by marcelklehr** [comment]
This should be resolved in version 4.0.0, please shout if you feel it isn't :)
